aiida/backends/tests/test_nodes.py

```diff
@@ -1702,100 +1621,385 @@ def test_deletion_non_existing_pk(self):
+        self._check_existence(uuids_check_existence, uuids_check_deleted)
 
-        uuids_check_existence = (in1.uuid,)
-        uuids_check_deleted = [n.uuid for n in (wf, in2)]
+        # If you want to keep the independent process, first delete the global
+        # workflow
+        dia, doa, pca, pwa, dib, dob, pcb, pwb, pw0 = self._create_indep2w_graph()
 
+        uuids_check_existence = [n.uuid for n in [dia, doa, pca, pwa, dib, dob, pcb, pwb]]
+        uuids_check_deleted = [n.uuid for n in [pw0]]
         with Capturing():
-            delete_nodes([wf.pk], verbosity=2, force=True, follow_returns=True)
+            delete_nodes((pw0.pk,), verbosity=2, force=True, call_calc_forward=False, call_work_forward=False)
+        self._check_existence(uuids_check_existence, uuids_check_deleted)
 
+        uuids_check_existence = [n.uuid for n in [dia, dib, dob, pcb, pwb]]
+        uuids_check_deleted = [n.uuid for n in [doa, pca, pwa]]
+        with Capturing():
+            delete_nodes((pwa.pk,), verbosity=2, force=True, call_calc_forward=True, call_work_forward=True)
         self._check_existence(uuids_check_existence, uuids_check_deleted)
 
-    def test_delete_called_but_not_caller(self):
+
+    def _create_long_graph(self):
         """
-        Check that deleting a ProcessNode that was called by another ProcessNode which won't be
-        deleted works, even though it will raise a warning
+        Creates a straighforward graph with up to N=20 (currently) nodes. It
+        checks if propagation works correctly, but it also can be used quickly
+        to make preliminary timing profiling (increasing hardcoded N). The link
+        scheeme looks like this::
+
+            +----+     +----+     +----+     +----+     +----+
+            | D0 | --> | C1 | --> | D1 | --> | C2 | --> | D2 | --> ...
+            +----+     +----+     +----+     +----+     +----+
         """
-        caller, called = [orm.WorkflowNode() for i in range(2)]
-        called.add_incoming(caller, link_type=LinkType.CALL_WORK, link_label='link')
-        caller.store()
-        called.store()
 
-        uuids_check_existence = (caller.uuid,)
-        uuids_check_deleted = [n.uuid for n in (called,)]
+        old_data = orm.Data().store()
+        node_list_id = [old_data.id]
 
-        with Capturing():
-            delete_nodes([called.pk], verbosity=2, force=True, follow_returns=True)
+        for ii in range(20):
+            new_calc = orm.CalculationNode()
+            new_data = orm.Data().store()
+            new_calc.add_incoming(old_data, link_type=LinkType.INPUT_CALC, link_label='inp' + str(ii))
+            new_calc.store()
+            new_data.add_incoming(new_calc, link_type=LinkType.CREATE, link_label='out' + str(ii))
+            node_list_id.append(new_calc.id)
+            node_list_id.append(new_data.id)
+            old_data = orm.load_node(new_data.id)
+
+        node_list = []
+        for node_id in node_list_id:
```

Comment:
also in this case, just create (above) `node_list` directly instead of creating a list of ids